org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/AsyncCompletionProposalPopup.java

```diff
@@ -382,7 +382,11 @@ protected List<CompletableFuture<List<ICompletionProposal>>> buildCompletionFutu
 					pool -> new ForkJoinWorkerThread(pool) {
 						// anonymous subclass to access protected constructor
 					}, null, false);
-			futures.add(CompletableFuture.supplyAsync(() -> this.getCompletionProposals(processor, invocationOffset), commonPool));
+			try {
```

Comment:
Does it make sense to create and shutdown thread pool **per proposal**?

That should be done once for all!